main.cc

```diff
@@ -162,10 +167,10 @@ int main(int argc, char* argv[])
 #ifdef timeit
     unsigned long endtime = CLOCK();
     unsigned long nidend = NODE_ID();
-    if (nid != nidend) printf("timing problem %lu %lu\n", nid, nidend);
+    if (nid != nidend) printf("\n Timing problem %lu %lu \n", nid, nidend);
     unsigned long totaltime = endtime - starttime;
     double ms = ((double) totaltime / clockrate) / 1000.0;
-    printf("Clock %.1lf Mhz\t Total Cycles %lu\t Time(ms) %.1lf\n",
+    printf("ClockRate %.1lf Mhz\t Total Cycles %lu\t Time(ms) %.1lf\n",
          clockrate, totaltime, ms); fflush(stdout);
 #endif
```

Comment:
Please put the `timeit` stuff into subroutines that appear above main so that main stays as clean as possible.

Reply:
replaced timeit stuff with timing_hooks in the next commit, which also contains starttiming.